scipy/spatial/qhull.pyx

```diff
@@ -494,6 +510,8 @@ cdef class _Qhull:
                 qh_setdelaunay(self._qh, arr.shape[1], arr.shape[0], <realT*>arr.data)
 
             p = <realT*>arr.data
+            if self._is_halfspaces:
+                p = qh_sethalfspace_all(self._qh, arr.shape[1], arr.shape[0], p, self._qh.feasible_point)
```

Comment:
You are right, it should be freed: I forgot about it because in the other case it always references the ndarray...